libraries/chain/include/eosio/chain/backing_store/db_key_value_iter_cache.hpp

```diff
@@ -0,0 +1,156 @@
+#pragma once
+#include <fc/utility.hpp>
+#include <sstream>
+#include <algorithm>
+#include <set>
+
+namespace eosio { namespace chain { namespace backing_store {
+struct unique_table {
+   name contract;
+   name scope;
+   name table;
+};
+
+template<typename T>
+struct secondary_key {
+   int       table_ei;
+   T         secondary;
+   uint64_t  primary;
+   uint64_t  payer;
+};
+
+struct unique_table_hash {
```

Comment:
I need my memory refreshed on this, but couldn't we add a partial specialization of `std::hash`?  I thought the standard allowed for partial specialization of class templates as long as it depended on a user defined type.

Reply:
dropping hash and switching to map, so this is a moot point